drivers/accel/adxl313/iio_adxl313.c

```diff
@@ -0,0 +1,1140 @@
+		.show  = adxl313_iio_read_calibbias,
+		.store = adxl313_iio_write_calibbias,
+	},
+	{
+		.name = "raw",
+		.show = adxl313_iio_read_raw,
+	},
+	{
+		.name   = "sampling_frequency",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl313_iio_read_sampling_freq,
+		.store  = adxl313_iio_write_sampling_freq,
+	},
+	{
+		.name   = "sampling_frequency_available",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl313_iio_read_samp_freq_avail,
+	},
+	{
+		.name   = "range",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl313_iio_read_range,
+		.store  = adxl313_iio_write_range,
+	},
+	{
+		.name   = "range_available",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl313_iio_read_range_avail,
+	},
+	{
+		.name   = "scale",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl313_iio_read_scale,
+		.store  = adxl313_iio_write_scale,
+	},
+	{
+		.name   = "scale_available",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl313_iio_read_scale_avail,
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct scan_type adxl313_iio_accel_scan_type = {
+	.sign = 's',
+	.realbits = 13,
+	.storagebits = 16,
+	.shift = 0,
+	.is_big_endian = false
+};
+
+#define ADXL313_ACCEL_CHANNEL(index, reg, axis) { \
+ 	.ch_type = IIO_ACCEL,                         \
+ 	.channel = index,                             \
+ 	.address = reg,                               \
+ 	.modified = true,                             \
+ 	.channel2 = IIO_MOD_##axis,                   \
+ 	.scan_type = &adxl313_iio_accel_scan_type,    \
+ 	.scan_index = index,                          \
+ 	.attributes = adxl313_iio_accel_attrs,        \
+ 	.ch_out = false                               \
+ }
+
+static struct iio_channel adxl313_channels[] = {
+	ADXL313_ACCEL_CHANNEL(0, chan_x, X),
+	ADXL313_ACCEL_CHANNEL(1, chan_y, Y),
+	ADXL313_ACCEL_CHANNEL(2, chan_z, Z),
+};
+
+static struct iio_device adxl313_iio_dev = {
+	.num_ch = NO_OS_ARRAY_SIZE(adxl313_channels),
+	.channels = adxl313_channels,
+	.pre_enable = (int32_t (*)())adxl313_iio_update_channels,
+	.read_dev = (int32_t (*)())adxl313_iio_read_samples,
+	.debug_reg_read = (int32_t (*)())adxl313_iio_read_reg,
+	.debug_reg_write = (int32_t (*)())adxl313_iio_write_reg
+};
+
+/*******************************************************************************
+ * @brief Searches for match in the given 2-row table
+ *
+ * @param freq_tbl - The frequency table in which the search is performed
+ * @param size     - The line index until which the search is performed
+ * @param val1     - Value on the first row to be checked
+ * @param val2     - Value on the second row to be checked
+ *
+ * @return ret     - Table index which contains the searched data
+*******************************************************************************/
+static int adxl313_iio_find_2d_row(const int (*freq_tbl)[2], const int size,
+				   const int val1, const int val2)
+{
+	int i;
+
+	for (i = 0; i < size; i++) {
+		if (freq_tbl[i][0] == val1 && freq_tbl[i][1] == val2)
```

Comment:
hmm, i think brackets should be used here:

`((freq_tbl[i][0] == val1) && (freq_tbl[i][1] == val2))`